src/wwwroot/template-editor/adapterService.js

```diff
@@ -354,15 +356,27 @@
      */
     function getCampaign(id, useEditorAsTemplate) {
       traceAdapterCall(getCampaign, arguments);
-      var campaign = {
-        id: id,
-        type: 'campaign',
-        name: 'Campaign ' + id,
-        attributes: {},
-        innerHTML: '',
-        children: []
-      };
-      return $q.resolve({ data: campaign });
+      return $http.get(`${RELAY_CONFIG.baseUrl}/accounts/${loginSession.accountId}/template/${id}`, {
+        params: {
+          useAsTemplate: useEditorAsTemplate
+        },
+        headers: {
+          'Authorization': 'Bearer '+ apiToken
+        }
+      }).then(function(response) {
+        var campaign = {
+          id: id,
+          type: 'campaign',
+          name: 'Campaign ' + id,
+          attributes: {},
+          innerHTML: '',
+          children: []
+        };
+        return { data: campaign };
+      },
+      function(error) {
+        $window.location = "/#/templates"
```

Comment:
I am not sure about it. What do you think about a popup or an alert and then redirect or refresh?

Reply:
Ok, I will add an alert describing the error.